fluid/neural_machine_translation/transformer/train.py

```diff
@@ -170,32 +186,57 @@ def test(exe):
 
     # Initialize the parameters.
     exe.run(fluid.framework.default_startup_program())
+
+    data_input_names = encoder_data_input_fields + decoder_data_input_fields[:
+                                                                             -1] + label_data_names
+    util_input_names = encoder_util_input_fields + decoder_util_input_fields
+
+    train_exe = fluid.ParallelExecutor(
+        use_cuda=TrainTaskConfig.use_gpu,
+        loss_name=avg_cost.name
+        if TrainTaskConfig.use_avg_cost else sum_cost.name)
+
+    dev_count = fluid.core.get_cuda_device_count()
+
     for pos_enc_param_name in pos_enc_param_names:
-        pos_enc_param = fluid.global_scope().find_var(
-            pos_enc_param_name).get_tensor()
-        pos_enc_param.set(
-            position_encoding_init(ModelHyperParams.max_length + 1,
-                                   ModelHyperParams.d_model), place)
+        tensor = position_encoding_init(ModelHyperParams.max_length + 1, ModelHyperParams.d_model)
+        for place_id in xrange(dev_count):
```

Comment:
Maybe we can have 2 arguments for ParallelExecutor.run(), one called feed_data_parallel_dict and feed_dict. feed_data_parallel_dict will split the tensor at the batch dimension while feed_dict will replicate the data to each place.
The logics are implemented in C++ so that it's simpler for the users.

Another question is, if the var being fed will be needed in parameter servers, such as learning rate. will feed_dict works?